### ISSUE TITLE
MSpanList_Insert error when calling mulled-build build-and-test

### ISSUE THREAD

**Issue by pcm32** [galaxyproject/galaxy-lib#124]
Hi there,

I'm trying the `mulled-build build-and-test` to build a container locally from a conda package through:

```
mulled-build build-and-test 'r-seurat-workflow=0.0.1--0' --extra-channel file://Users/pmoreno/miniconda3/conda-bld/ --test 'which read-10x.R'
```

but getting an error:

```
./involucro -v=2 -f /Users/pmoreno/venvs/mulled/lib/python2.7/site-packages/galaxy/tools/deps/mulled/invfile.lua -set CHANNELS='bioconda,file://Users/pmoreno/miniconda3/conda-bld/' -set TEST='which read-10x.R' -set TARGETS='r-seurat-workflow=0.0.1=0' -set REPO='quay.io/biocontainers/r-seurat-workflow:0.0.1--0' -set BINDS='build/dist:/usr/local/,/Users/pmoreno/miniconda3/conda-bld/:/Users/pmoreno/miniconda3/conda-bld/' build-and-test
.No handlers could be found for logger "galaxy.tools.deps.commands"
failed MSpanList_Insert 0x878000 0x1f3fa463031a7 0x0 0x0
fatal error: MSpanList_Insert

runtime stack:
runtime.throw(0x584810, 0x10)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/panic.go:530 +0x90 fp=0x7ffeefbff310 sp=0x7ffeefbff2f8
runtime.(*mSpanList).insert(0x7bd888, 0x878000)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/mheap.go:933 +0x293 fp=0x7ffeefbff340 sp=0x7ffeefbff310
runtime.(*mheap).freeSpanLocked(0x7bd080, 0x878000, 0x100, 0x0)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/mheap.go:809 +0x4be fp=0x7ffeefbff3a8 sp=0x7ffeefbff340
runtime.(*mheap).grow(0x7bd080, 0x8, 0x0)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/mheap.go:675 +0x2a0 fp=0x7ffeefbff400 sp=0x7ffeefbff3a8
runtime.(*mheap).allocSpanLocked(0x7bd080, 0x1, 0x0)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/mheap.go:553 +0x4e3 fp=0x7ffeefbff458 sp=0x7ffeefbff400
runtime.(*mheap).alloc_m(0x7bd080, 0x1, 0x15, 0x0)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/mheap.go:437 +0x119 fp=0x7ffeefbff488 sp=0x7ffeefbff458
runtime.(*mheap).alloc.func1()
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/mheap.go:502 +0x41 fp=0x7ffeefbff4b8 sp=0x7ffeefbff488
runtime.systemstack(0x7ffeefbff4d8)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/asm_amd64.s:307 +0xab fp=0x7ffeefbff4c0 sp=0x7ffeefbff4b8
runtime.(*mheap).alloc(0x7bd080, 0x1, 0x10000000015, 0xe9bf)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/mheap.go:503 +0x63 fp=0x7ffeefbff508 sp=0x7ffeefbff4c0
runtime.(*mcentral).grow(0x7bec80, 0x0)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/mcentral.go:209 +0x93 fp=0x7ffeefbff570 sp=0x7ffeefbff508
runtime.(*mcentral).cacheSpan(0x7bec80, 0x7b7b08)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/mcentral.go:89 +0x47d fp=0x7ffeefbff5b0 sp=0x7ffeefbff570
runtime.(*mcache).refill(0x874000, 0x15, 0x7ffeefbff618)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/mcache.go:119 +0xcc fp=0x7ffeefbff5e8 sp=0x7ffeefbff5b0
runtime.mallocgc.func2()
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/malloc.go:642 +0x2b fp=0x7ffeefbff608 sp=0x7ffeefbff5e8
runtime.systemstack(0x7ffeefbff6a8)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/asm_amd64.s:307 +0xab fp=0x7ffeefbff610 sp=0x7ffeefbff608
runtime.mallocgc(0x180, 0x521e60, 0x0, 0x800000000)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/malloc.go:643 +0x869 fp=0x7ffeefbff6e8 sp=0x7ffeefbff610
runtime.newobject(0x521e60, 0x7b7fb0)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/malloc.go:781 +0x42 fp=0x7ffeefbff710 sp=0x7ffeefbff6e8
runtime.malg(0x8000, 0x7b8360)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/proc.go:2634 +0x27 fp=0x7ffeefbff748 sp=0x7ffeefbff710
runtime.mpreinit(0x7b8900)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/os1_darwin.go:140 +0x1f fp=0x7ffeefbff760 sp=0x7ffeefbff748
runtime.mcommoninit(0x7b8900)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/proc.go:494 +0x105 fp=0x7ffeefbff7a8 sp=0x7ffeefbff760
runtime.schedinit()
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/proc.go:434 +0x79 fp=0x7ffeefbff7f0 sp=0x7ffeefbff7a8
runtime.rt0_go(0x7ffeefbff820, 0xf, 0x7ffeefbff820, 0x0, 0xf, 0x7ffeefbff9b8, 0x7ffeefbff9c4, 0x7ffeefbff9c9, 0x7ffeefbff9cc, 0x7ffeefbffa29, ...)
	/Users/travis/.gimme/versions/go1.6.darwin.amd64/src/runtime/asm_amd64.s:138 +0x132 fp=0x7ffeefbff7f8 sp=0x7ffeefbff7f0
```

This was both using the staticly built involucro that I manually downloaded from the releases on github or relying on the auto download (removing my binary from the path). Should I be trying to build involucro locally?

Before trying this I conda built my package locally succesfully (and indexed my local lib). I installed galaxy-lib through `pip install galaxy-lib`.

Any ideas @bgruening?

Thanks!

**Comment by pcm32** [comment]
I see that this has been reported here as well:
https://github.com/bioconda/bioconda-recipes/issues/8639

**Comment by pcm32** [comment]
Will try re-compiling locally involucro, as there are many similar errors posted on Github on 2016, when this was compiled...

**Comment by pcm32** [comment]
The error above is resolved by using a self compiled version of involucro (which wasn't straightforward based on involucro documentation; I had to do `go build cmd/involucro/main.go` inside the involucro git clone and then rename the produced main). I had to call mulled-build with additional argument `--involucro-path <my-path-to-new-binary>/involucro` for the python code to pick it up (this should be added to the read the docs documentation probably). With all of these, I'm now getting issues with the fact that the package is local.... will probably open a different issue for that.
